app/helpers/injection_helper.rb

```diff
@@ -2,7 +2,7 @@
 
 module InjectionHelper
   def inject_enterprises
-    inject_json_ams "enterprises", Enterprise.activated.includes(:address).all, Api::EnterpriseSerializer, enterprise_injection_data
+    inject_json_ams "enterprises", Enterprise.activated.includes(address: :state).all, Api::EnterpriseSerializer, enterprise_injection_data
```

Comment:
Just to add - If you use `.explain` instead of `.to_sql` in your statements above you should see a difference. Also if you use irb (i.e. `rails c`,  pry suppresses the SQL by default) you can see fewer queries:

```{ruby}
ents = Enterprise.activated.includes(:address).all
ents.map{|e| e.address.state}
# => one query per address

entstate = Enterprise.activated.includes(address: :state).all
entstate.map{|e| e.address.state}
# => no further queries
```

This enterprise injection runs on every page so I guess it's good if we can improve the performance - I didn't test this directly though, just added it for the sake of good practice.